### DISCUSSION TITLE
rpc.tcp_server_port, rpc.ssl_server_port, and rpc.tcp_client_num

### DISCUSSION THREAD

**Discussion by tigercl** [emqx/emqx#13410] [General]
`rpc.tcp_server_port` and `rpc.ssl_server_port` are completely mutually exclusive. Can they be combined into one configuration item? For example, `rpc.server_port`.

`rpc.tcp_client_num` seems to do this. It also works when TLS is enabled, but I don't know why it has a `tcp_` prefix. Would it be more appropriate to change it to `rpc.client_num`?

**Comment by zmstone** [comment]
> rpc.tcp_server_port and rpc.ssl_server_port are completely mutually exclusive. Can they be combined into one configuration item? For example, rpc.server_port.

True. ~Separating the two config allows testing two nodes, and exactly two nodes in one host.~

**Comment by zmstone** [comment]
> rpc.tcp_client_num seems to do this. It also works when TLS is enabled, but I don't know why it has a tcp_ prefix. Would it be more appropriate to change it to rpc.client_num?

Agree, can use a more generic name.
Will rename to `clients_num`, and keep `tcp_client_num` as alias.

**Comment by tigercl** [comment]
So we will get the two configuration items: `server_port` and `clients_num`?

**Comment by zmstone** [comment]
https://github.com/emqx/emqx/pull/13434
